[broadcom] saibcm-modules-legacy-th build fails: stale symlinks from prior kernel module builds

## Description

Building `sonic-aboot-broadcom.swi` from a clean clone of current master (`32875b5f8`) fails when building `opennsl-modules-legacy-th`:

```
ln: failed to create symbolic link '.../arch/x86/module.lds': File exists
make[1]: *** [Makefile:15: /sonic/target/debs/bookworm/frr_10.5.1-sonic-0_amd64.deb] Error 128
```

Actually the error is:
```
ln: failed to create symbolic link '/usr/src/linux-headers-6.12.41+deb13-common-sonic/arch/x86/module.lds': File exists
```

## Root Cause

Three Broadcom kernel module packages build sequentially via `_AFTER` ordering:

1. **saibcm-modules** (first) — creates symlinks in `/usr/src/linux-headers-*-common-sonic/` with bare `ln -s`
2. **saibcm-modules-dnx** (second) — has `if [ ! -e ... ]` guards, so it skips existing symlinks ✅
3. **saibcm-modules-legacy-th** (third) — uses bare `ln -s` without guards, **fails** because symlinks already exist from step 1 ❌

The `saibcm-modules-legacy-th` directory was introduced in PR #25192 (as `saibcm-modules-legacy`) by copying from `saibcm-modules` without adding the existence guards that `saibcm-modules-dnx` has. PR #25871 renamed it to `legacy-th` but did not fix the bug.

## Affected Files

- `platform/broadcom/saibcm-modules/debian/rules` (lines 96-99) — no guards, but builds first so it works
- `platform/broadcom/saibcm-modules-dnx/debian/rules` (lines 96-107) — has `if [ ! -e ]` guards ✅
- `platform/broadcom/saibcm-modules-legacy-th/debian/rules` (lines 96-99) — no guards ❌

## Proposed Fix

Use `ln -sf` (force) instead of `ln -s` in all three `debian/rules` files. This is:
- Simpler than `if [ ! -e ]` guards
- Idempotent and defensive
- Standard practice in kernel module Makefiles (Linux's own `scripts/Makefile.build` uses `ln -sf`)

## Steps to Reproduce

```bash
git clone --recurse-submodules https://github.com/sonic-net/sonic-buildimage.git
cd sonic-buildimage
make init
make configure PLATFORM=broadcom
make target/sonic-aboot-broadcom.swi
```

Build fails at `opennsl-modules-legacy-th` every time.

## Impact

**All broadcom aboot/onie image builds are broken on current master** — this is a guaranteed failure since the `_AFTER` ordering ensures `saibcm-modules` always creates the symlinks before `saibcm-modules-legacy-th` tries to.

/cc @lipxu

--- a/platform/broadcom/saibcm-modules-legacy-th/debian/rules
+++ b/platform/broadcom/saibcm-modules-legacy-th/debian/rules
@@ -89,14 +89,12 @@ build-arch-stamp:
 
 	# create links
 	cd /; sudo mkdir -p /lib/modules/$(KVER_ARCH)
-	cd /; sudo rm /lib/modules/$(KVER_ARCH)/build
-	cd /; sudo rm /lib/modules/$(KVER_ARCH)/source
-	cd /; sudo ln -s /usr/src/linux-headers-$(KVER_COMMON)/ /lib/modules/$(KVER_ARCH)/source
-	cd /; sudo ln -s /usr/src/linux-headers-$(KVER_ARCH)/ /lib/modules/$(KVER_ARCH)/build
-	cd /; sudo ln -s /usr/src/linux-headers-$(KVER_ARCH)/include/generated/ /usr/src/linux-headers-$(KVER_COMMON)/include/generated
-	cd /; sudo ln -s /usr/src/linux-headers-$(KVER_ARCH)/arch/x86/include/generated/ /usr/src/linux-headers-$(KVER_COMMON)/arch/x86/include/generated
-	cd /; sudo ln -s /usr/src/linux-headers-$(KVER_ARCH)/arch/x86/module.lds /usr/src/linux-headers-$(KVER_COMMON)/arch/x86/module.lds
-	cd /; sudo ln -s /usr/src/linux-headers-$(KVER_ARCH)/include/config/ /usr/src/linux-headers-$(KVER_COMMON)/include/config
+	cd /; sudo ln -sfn /usr/src/linux-headers-$(KVER_COMMON)/ /lib/modules/$(KVER_ARCH)/source
+	cd /; sudo ln -sfn /usr/src/linux-headers-$(KVER_ARCH)/ /lib/modules/$(KVER_ARCH)/build
+	cd /; sudo ln -sfn /usr/src/linux-headers-$(KVER_ARCH)/include/generated/ /usr/src/linux-headers-$(KVER_COMMON)/include/generated
+	cd /; sudo ln -sfn /usr/src/linux-headers-$(KVER_ARCH)/arch/x86/include/generated/ /usr/src/linux-headers-$(KVER_COMMON)/arch/x86/include/generated
+	cd /; sudo ln -sf /usr/src/linux-headers-$(KVER_ARCH)/arch/x86/module.lds /usr/src/linux-headers-$(KVER_COMMON)/arch/x86/module.lds
+	cd /; sudo ln -sfn /usr/src/linux-headers-$(KVER_ARCH)/include/config/ /usr/src/linux-headers-$(KVER_COMMON)/include/config
 	cd /; sudo cp /usr/src/linux-headers-$(KVER_ARCH)/Module.symvers /usr/src/linux-headers-$(KVER_COMMON)/Module.symvers
 
 	# Add here command to compile/build the package.

--- a/platform/broadcom/saibcm-modules/debian/rules
+++ b/platform/broadcom/saibcm-modules/debian/rules
@@ -89,14 +89,12 @@ build-arch-stamp:
 
 	# create links
 	cd /; sudo mkdir -p /lib/modules/$(KVER_ARCH)
-	cd /; sudo rm /lib/modules/$(KVER_ARCH)/build
-	cd /; sudo rm /lib/modules/$(KVER_ARCH)/source
-	cd /; sudo ln -s /usr/src/linux-headers-$(KVER_COMMON)/ /lib/modules/$(KVER_ARCH)/source
-	cd /; sudo ln -s /usr/src/linux-headers-$(KVER_ARCH)/ /lib/modules/$(KVER_ARCH)/build
-	cd /; sudo ln -s /usr/src/linux-headers-$(KVER_ARCH)/include/generated/ /usr/src/linux-headers-$(KVER_COMMON)/include/generated
-	cd /; sudo ln -s /usr/src/linux-headers-$(KVER_ARCH)/arch/x86/include/generated/ /usr/src/linux-headers-$(KVER_COMMON)/arch/x86/include/generated
-	cd /; sudo ln -s /usr/src/linux-headers-$(KVER_ARCH)/arch/x86/module.lds /usr/src/linux-headers-$(KVER_COMMON)/arch/x86/module.lds
-	cd /; sudo ln -s /usr/src/linux-headers-$(KVER_ARCH)/include/config/ /usr/src/linux-headers-$(KVER_COMMON)/include/config
+	cd /; sudo ln -sfn /usr/src/linux-headers-$(KVER_COMMON)/ /lib/modules/$(KVER_ARCH)/source
+	cd /; sudo ln -sfn /usr/src/linux-headers-$(KVER_ARCH)/ /lib/modules/$(KVER_ARCH)/build
+	cd /; sudo ln -sfn /usr/src/linux-headers-$(KVER_ARCH)/include/generated/ /usr/src/linux-headers-$(KVER_COMMON)/include/generated
+	cd /; sudo ln -sfn /usr/src/linux-headers-$(KVER_ARCH)/arch/x86/include/generated/ /usr/src/linux-headers-$(KVER_COMMON)/arch/x86/include/generated
+	cd /; sudo ln -sf /usr/src/linux-headers-$(KVER_ARCH)/arch/x86/module.lds /usr/src/linux-headers-$(KVER_COMMON)/arch/x86/module.lds
+	cd /; sudo ln -sfn /usr/src/linux-headers-$(KVER_ARCH)/include/config/ /usr/src/linux-headers-$(KVER_COMMON)/include/config
 	cd /; sudo cp /usr/src/linux-headers-$(KVER_ARCH)/Module.symvers /usr/src/linux-headers-$(KVER_COMMON)/Module.symvers
 
 	# Add here command to compile/build the package.

